modules/geoedgeRtdProvider.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * This module adds geoedge provider to the real time data module
+ * The {@link module:modules/realTimeData} module is required
+ * The module will fetch creative wrapper from geoedge server
+ * The module will place geoedge RUM client on bid responses markup
+ * @module modules/geoedgeProvider
+ * @requires module:modules/realTimeData
+ */
+
+/**
+ * @typedef {Object} ModuleParams
+ * @property {string} key
+ * @property {?Object} bidders
+ * @property {?boolean} wap
+ * @property {?string} keyName
+ */
+
+import { submodule } from '../src/hook.js';
+import { ajax } from '../src/ajax.js';
+import { generateUUID, insertElement, isEmpty, logError } from '../src/utils.js';
+
+/** @type {string} */
+const SUBMODULE_NAME = 'geoedge';
+/** @type {string} */
+export const WRAPPER_URL = '//wrappers.geoedge.be/wrapper.html';
```

Comment:
Ok, structure of the module is looking better. Will assign someone to look deeper at the javascript. Now we have to talk about this external code. In order to protect publisher pages, [Global Module Rule #1](https://docs.prebid.org/dev-docs/module-rules.html#global-module-rules) is `No External Code`. Looking at wrapper.html, it's pretty small, but it loads another file: `https://rumcdn.geoedge.be/$%7Bkey%7D/grumi.js`.

You're going to have to figure out how to live within this Module Rule, basically, all code that's loaded must be inspectable, disclosed, and locked to a specific version so we know when you've changed it.  Since the whole point of GeoEdge is to help protect publishers, I suspect you'll understand why this rule exists. :-)

If wrapper.html may be more-or-less static, you might consider including that right in this code. grumi.js probaby depends on the publisher, so for that you're going to need to A) open source it and B) lock it to a version, e.g. grumi-v1.js

If grumi.js needs data (e.g. malware patterns), you might consider: 
- separate the code from the data. Loading data files is fine.
- make grumi.js as static as possible and put your application logic server-side